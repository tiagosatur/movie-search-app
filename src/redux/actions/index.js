export * from './movies';
export * from './details';