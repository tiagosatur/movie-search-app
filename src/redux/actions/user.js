import { 
    LOGIN_USER_PENDING,
    LOGIN_USER_SUCCESS,
    LOGIN_USER_ERROR,
} from '../../utils/actionTypes';

function loginUserPending() {
    return {
        type: LOGIN_USER_PENDING
    }
}

export function loginUserSuccess(user) {    
    return {
        type: LOGIN_USER_SUCCESS,
        user
    }
}

function loginUserError(error) {
    return {
        type: LOGIN_USER_ERROR,
        error: error
    }
}

export const loginUser = (data) => (dispatch) => {
    dispatch(loginUserSuccess(data))
}