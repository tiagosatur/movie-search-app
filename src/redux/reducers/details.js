import { 
    MOVIE_DETAILS_PENDING,
    MOVIE_DETAILS_SUCCESS,
    MOVIE_DETAILS_ERROR,
} from '../../utils/actionTypes';

export const initialState = {
    pending: false,
    details: {},
    error: null
}

export default ( state = initialState, action ) => {
    const { type, details, error } = action;

    switch(type) {
        case MOVIE_DETAILS_PENDING:
            return {
                ...state,
                details: {},
                pending: true
            };
        case MOVIE_DETAILS_SUCCESS:
            return {
                ...state,
                pending: false,
                details: details,
            };
        case MOVIE_DETAILS_ERROR:
            return {
                ...state,
                details: {},
                pending: false,
                error: error
            };
        default: 
            return state;
    }
    
};