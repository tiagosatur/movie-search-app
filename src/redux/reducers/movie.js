import { 
    SEARCH_MOVIE_PENDING,
    SEARCH_MOVIE_SUCCESS,
    SEARCH_MOVIE_ERROR,
    SEARCH_MOVIE_CLEAR,
} from '../../utils/actionTypes';

export const initialState = {
    pending: false,
    movie: [],
    totalResults: 0,
    error: null
}

export default ( state = initialState, action ) => {
    const { type, movie, error } = action;

    switch(type) {
        case SEARCH_MOVIE_PENDING:
            return {
                ...state,
                movie: [],
                totalResults: 0,
                pending: true
            };
        case SEARCH_MOVIE_CLEAR :
            return {
                ...state,
                movie: [],
                totalResults: 0,
                pending: true
            };
        case SEARCH_MOVIE_SUCCESS:
            return {
                ...state,
                pending: false,
                movie: movie.Search,
                totalResults: movie.totalResults,
            };
        case SEARCH_MOVIE_ERROR:
            return {
                ...state,
                movie: [],
                pending: false,
                totalResults: 0,
                error: error
            };
        default: 
            return state;
    }
};