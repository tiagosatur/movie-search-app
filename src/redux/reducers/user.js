import { 
    LOGIN_USER_PENDING,
    LOGIN_USER_SUCCESS,
    LOGIN_USER_ERROR,
} from '../../utils/actionTypes';

export const initialState = {
    pending: false,
    isLogged: false,
    user: {},
    error: null
}

export const user = ( state = initialState, action ) => {
    const { type, user, error } = action;

    switch(type) {
        case LOGIN_USER_PENDING:
            return {
                ...state,
                user: {},
                pending: true
            };
        case LOGIN_USER_SUCCESS:
            return {
                ...state,
                pending: false,
                isLogged: true,
                user: user,
            };
        case LOGIN_USER_ERROR:
            return {
                ...state,
                pending: false,
                isLogged: false,
                user: {},
                error: error
            };
        default: 
            return state;
    }
};