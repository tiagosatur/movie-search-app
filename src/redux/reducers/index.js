import { combineReducers } from 'redux'
import {movie} from './movie'
import {details} from './details'
import {user} from './user'

export default combineReducers ({
    movie,
    details,
    user,
})