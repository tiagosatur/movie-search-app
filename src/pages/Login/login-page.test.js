import React from 'react';
import {  Provider } from 'react-redux'
import { initialState } from '../../redux/reducers/user';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ConnectedLogin, { Login } from './Login';
import { MemoryRouter } from "react-router";
import { createStore } from 'redux';
import rootReducer from '../../redux/reducers'
import { spy } from 'sinon';
import { act } from 'react-dom/test-utils';

describe('Login Page tests', () => {
    let store = createStore(rootReducer, initialState);
    const callback = spy();

    const x = shallow(
        <Provider store={store}>
            <MemoryRouter initialEntries={[ '/login' ]}>
                <Login />
            </MemoryRouter>
        </Provider>
    ).dive();
    
    const wrapper = mount(
        <Provider store={store}>
            <MemoryRouter initialEntries={[ '/login' ]}>
                <ConnectedLogin />
            </MemoryRouter>
        </Provider>
    );

    it('should render the Search Form', () => {
        expect(wrapper.find('.login-page').exists()).toBe(true);
    });
    
    it('should call Login Page snapshot correctly', () => {
        const tree = renderer
            .create(wrapper)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should call onSubmit function when form is submitted', () => {
        const form = wrapper.find('form').at(0);
        const children = form.render().children().children();
        form.simulate('submit', { target: { children } });
    });

    it('should have two input fields', () => {
        expect(wrapper.find('input').length).toBe(2);
    });

    it('should have one submit button', () => {
        expect(wrapper.find('button').length).toBe(1);
    });
      
})
