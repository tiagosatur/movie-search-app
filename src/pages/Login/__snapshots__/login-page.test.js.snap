// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page tests should call Login Page snapshot correctly 1`] = `
<div
  className="login-page"
>
  <p
    className="sc-jzJRlG kIVgys"
  >
    You should be logged to access the page
    <strong>
       
      /
       
    </strong>
  </p>
  <div>
    <form
      className="login-form sc-fjdhpX kNBbKB"
      onSubmit={[Function]}
    >
      <div
        className="sc-htpNat bqYrTX"
      >
        <input
          className="input  sc-bxivhb izmEcM"
          id="login-input-username"
          name="username"
          placeholder="Username"
          type="text"
        />
      </div>
      <div
        className="sc-htpNat bqYrTX"
      >
        <input
          className="input  sc-bxivhb izmEcM"
          id="login-input-password"
          name="password"
          placeholder="Password"
          type="password"
        />
      </div>
      <button
        className="button  sc-bdVaJa jllYbe"
        id="login-form-btn"
        title=""
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
